Update libmesh, add clang container

milljm opened the pull request:
Install clang with dnf. Install matching gfortran version that coincides with the version of gcc that gets installed as a dependency from installing clang.

Set necessary /opt/rh/ toolchain paths. This was a key piece of information I missed as the dependencies above were being installed. `clang` gets installed to /usr/bin/, while the dependecies in order to properly make use of said clang are installed in /opt/rh/etc (some path that is not available by default like /usr/bin is). Hence, confusion.

Updates libmesh; summary of changes:

- Added low-order `RAVIART_THOMAS` finite element support, including
  general H(div) mapping support for future divergence-conforming
  finite element families.  Added `L2_RAVIART_THOMAS` support for the
  broken/L2 version of that space.  Example codes for each, using mixed
  and hybridized formulations.
- Option to remove a boundary id globally from a mesh without
  re-preparing it afterward
- Option to merge subdomains when stitching meshes based on inspection
  of both subdomain ids and subdomain names
- Better matching of input file ids when reading Abaqus files
- Fixes for independent control over which variables see p-refinement
- Added `QBase::clone()` method
- Added parameter for user-supplied hash when doing Just-In-Time
  compilation with fparser
- Added `Packing` specializations for `Eigen` matrices; this enables
  parallel communication of these classes and of many derived types
  including them.
- Bug fixes for nodal interpolation of rational elements; this
  corrects an error in VTK output of many IsoGeometric Analysis mesh
  results.
- Bug fix for overzealous caching of rational elements' shape
  functions.  This corrects an error in computation on many
  IsoGeometric Analysis meshes
- Bug fixes for refinement of Tet14 elements.
- More robust `reduced_basis` EIM error indicator calculations
- Higher-order default quadrature choice in integration-based error
  estimators and error indicators, avoiding underintegration on
  superconvergent results
- Removal of Fortran/C++ detection for HDF5 libraries, so we can more
  easily find the desired C HDF5 library we do use.
- Upgrade to latest autotools versions in libMesh, TIMPI, and
  MetaPhysicL
- Update VTK compatibility to support VTK 9.3
- GCC warning workarounds to allow --enable-werror builds with -Wextra
  in GCC 13
- Minor refactoring in Threads/PerfLog interactions
- More parallel synchronization assertions during System setup and
  update operations
- Minor fixes to documentation, comments, and unit tests, including
  VexCL-enabled unit tests of MetaPhysicL

Commit:
Install clang from system package manager

Install clang with dnf. Install matching gfortran version that
coincides with the version of gcc that gets installed as a
dependency from installing clang.

Set necessary /opt/rh/ toolchain paths. This was a key piece of
information I missed as the dependencies above were being
installed. `clang` gets installed to /usr/bin/, while the dependecies
in order to properly make use of said clang are installed in
/opt/rh/etc (some path that is not available by default like
/usr/bin is). Hence, confusion.

Closes #25831

Commit:
Remove some duplication

Remove duplication, and unnecessary dnf install (libgfortran)

Commit:
Use Docker image

Try to use a docker image instead of yum mirror repo

moosebuild commented:
All [jobs](https://civet.inl.gov/event/151558/) on 63b89ae : canceled by @milljm

Commit:
Fix the dumb

milljm commented:
Monday: Warnings being treated as errors:

https://civet.inl.gov/job/1861845/
https://civet.inl.gov/job/1861844/

loganharbour committed:
Resolve clang warnings

loganharbour committed:
Resolve clang warnings in modules

loganharbour commented:
Well @milljm, after cleaning up those warnings... I think this is actually ready to go. even passed the clang tests!

Commit:
Remove double new lines

Remove double new lines as a separate commit...

Closes #25974

Commit:
Pass prefix options when WASP_PREFIX is set

Pass necessary config arguments to install wasp to a target location
if WASP_PREFIX is set.

Closes #25974

loganharbour committed:
Add lldb to clang container

roystgnr committed:
libMesh changelog in newsletter

Commit:
WASP Conda package bumps

Needed changes in WASP Conda package due to influential file.

Commit:
Bump MPICH (and the rest) due to mpich.def

Due to influential file: moose/apptainer/mpich.def, all packages
require an bump.

Will bump versioner_hashes in separate commit.

Merged WASP change into this PR to avoid a double moose-dev update.

Closes #25974 #25831

Commit:
Bump versioner_hashes

Bumping versioner_hashes due to package updates.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1865720/) on 641011a wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
This is now an "everything" PR... lets pay double attention to all the meta.yaml, etc bumps going on before hitting merge...

cticenhour commented:
How is 3a630b6 in this changeset not causing conflicts? Those changes are already in devel?

roystgnr commented:
>How is https://github.com/idaholab/moose/commit/3a630b6df89896a83336cc382f57ebfa0d1b503e in this changeset not causing conflicts? Those changes are already in devel.

Git is smart about not double-applying the same commit.

loganharbour committed:
Update libmesh submodule

Summary of changes:

- Added low-order RAVIART_THOMAS finite element support, including
  general H-div mapping support for future divergence-conforming
  finite element families.  Added L2_RAVIART_THOMAS support for the
  broken/L2 version of that space.  Example codes for each, using mixed
  and hybridized formulations.
- Option to remove a boundary id globally from a mesh without
  re-preparing it afterward
- Option to merge subdomains when stitching meshes based on inspection
  of both subdomain ids and subdomain names
- Better matching of input file ids when reading Abaqus files
- Fixes for independent control over which variables see p-refinement
- Added QBase::clone method
- Added parameter for user-supplied hash when doing Just-In-Time
  compilation with fparser
- Added  specializations for Eigen matrices; this enables
  parallel communication of these classes and of many derived types
  including them.
- Bug fixes for nodal interpolation of rational elements; this
  corrects an error in VTK output of many IsoGeometric Analysis mesh
  results.
- Bug fix for overzealous caching of rational elements shape
  functions.  This corrects an error in computation on many
  IsoGeometric Analysis meshes
- Bug fixes for refinement of Tet14 elements.
- More robust  EIM error indicator calculations
- Higher-order default quadrature choice in integration-based error
  estimators and error indicators, avoiding underintegration on
  superconvergent results
- Removal of Fortran/C++ detection for HDF5 libraries, so we can more
  easily find the desired C HDF5 library we do use.
- Upgrade to latest autotools versions in libMesh, TIMPI, and
  MetaPhysicL
- Update VTK compatibility to support VTK 9.3
- GCC warning workarounds to allow --enable-werror builds with -Wextra
  in GCC 13
- Minor refactoring in Threads/PerfLog interactions
- More parallel synchronization assertions during System setup and
  update operations
- Minor fixes to documentation, comments, and unit tests, including
  VexCL-enabled unit tests of MetaPhysicL

loganharbour committed:
Add update_and_rebuild_wasp.sh as a dep to wasp

milljm commented on scripts/tests/versioner_hashes.yaml:
```diff
@@ -153,3 +153,9 @@ d5e3e09267e1f2b47684a089150bec587178e1cf: #25784
   moose: 176f779
   wasp: 2fdee60
+286c130af84a0017140e6a5a910fb4bcfa47f8ae: #25831
+  mpic: e018192
```
LOL...

moosebuild commented on libmesh:
```diff
@@ -1 +1 @@
-Subproject commit ceff84386bdc83d87e9eb98ac3a9b745711e415a
+Subproject commit b2248b506847fd72d37cff56b6f4d8432f6c7e92
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866325/) on 56e8bb8 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866325/) on 56e8bb8 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866325/) on 56e8bb8 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866325/) on 56e8bb8 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

roystgnr commented:
C++ changes look good to me; I'm not confident enough in the apptainer/conda systems to hit approve on those changes.

If for some reason this doesn't get merged today I'll be pleading to advance the new libMesh hash to get a couple more upcoming changes there, but the libMesh update has been so delayed already that I don't want to delay it further on that account.

cticenhour commented on scripts/tests/versioner_hashes.yaml:
```diff
@@ -153,3 +153,9 @@ d5e3e09267e1f2b47684a089150bec587178e1cf: #25784
   libmesh: 50463c4
   moose: 176f779
   wasp: 2fdee60
+1cbf0084dc8ae0dde5f6a4500094f639eba55864: #25831
```
Isn't this comment usually the PR number, not the referenced issue?

cticenhour commented on conda/pyhit/meta.yaml:
```diff
@@ -1,5 +1,4 @@
-{% set build = 0 %}
-{% set version = "2023.10.24" %}
+{% set version = "2023.11.06" %}
```
`conda/pyhit/conda_build_config.yaml` still needs to be updated as well to `moose-wasp 2023.11.06`.

milljm replied:
👍  good catch!

Commit:
Add documentation, address requests

Add moose-dev version tracking to packages_config.yml. Updated each
instance where we are asking folks to perform a Conda update, to
instead delete and recreate their Conda moose environment. This is
unfortunately necessary, as I can reproduce Conda inability to update
our moose-dev package when specifically asking for that package to
be updated. Sad Face.

Fix versioner_hashes.yaml and pyhit's conda_build_config.yaml.

milljm commented on framework/doc/packages_config.yml:
```diff
@@ -6,3 +6,4 @@ minimum_python: 3.7
 mpich: 4.0.2
 petsc_alt: 3.11.4
 petsc: 3.16.6
```
@cticenhour I should probably change this too!

cticenhour replied:
Yes - we should while we're here. Good catch!

cticenhour replied:
Thanks for all the fixups @milljm!

Commit:
Bump petsc version in packages_config.yml

Bump petsc version in packages_config.yml and leave a hint about
doing so in the Conda meta.yaml file.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866604/) on fc693e9 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866604/) on fc693e9 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866604/) on fc693e9 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866604/) on fc693e9 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Add finally bump versioner_hashes...

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866704/) on 135d220 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

cticenhour reviewed (approved):
Review addressed. C++, conda, and documentation updates look reasonable to me. @loganharbour leaving one last look and any finishing requirements to you.

Commit:
Missed an instance of mamba update moose-dev

swap to delete env, create a new one method.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866958/) on 38fa98e wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866958/) on 38fa98e wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866958/) on 38fa98e wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1866958/) on 38fa98e wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

Commit:
Ugh. Mistake. Remove redundant tracking

Remove redundant tracking of update_and_rebuild_wasp.sh for moose.

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

milljm commented:
LOL... with all the pushes, I done broke the PR Conda channel somehow. Fixing... then invalidate all...

moosebuild commented:
All [jobs](https://civet.inl.gov/event/151945/) on 4c06c66 : invalidated by @milljm

Invalidate all after cleaning pr-25939 (missing moose-mpich package)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867084/) on 4c06c66 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

loganharbour committed:
Bump peacock package

loganharbour committed:
Add peacock influential package to moose due to moose-dev dependence

loganharbour committed:
Update versioner hashes

loganharbour commented:
We forgot about the fact that `moose-peacock` depends on `moose-mpich`. Bumped that. Version for moose didn't change, realized that we were also missing the influential files from `conda/peacock` for the `moose` version entry.

It never ends :-)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
All [jobs](https://civet.inl.gov/event/151978/) on 3f44eb1 : invalidated by @milljm

cleared PR channel for good measure...

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

A change of the following file(s) triggered this check:

  conda/petsc

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The conda build config was not changed but one or more of its dependencies have changed

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

The following file(s) are changed:

  conda/petsc/meta.yaml

The PETSc conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

A change of the following file(s) triggered this check:

  libmesh

The following file(s) are unchanged:

  conda/mpich/conda_build_config.yaml

The libmesh submodule or configuration was changed but the conda build config was not

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/1867873/) on 3f44eb1 wanted to post the following:

The following file(s) are changed:

  conda/libmesh/meta.yaml

The libmesh conda configuration has changed; ping @idaholab/moose-dev-ops

moosebuild commented:
Job [Test timings](https://civet.inl.gov/job/1867947/) on 3f44eb1 wanted to post the following:

View timings [here](https://mooseframework.inl.gov/docs/PRs/25939/timing)

This comment will be updated on new commits.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1867869/) on 3f44eb1 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25939/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1867909/) on 3f44eb1 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>29fb3b</th>
      <th colspan=4>#25939 3f44eb</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.67%</td>
      <td>85.67%</td>
      <td>-0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>92535</td>
      <td>92528</td>
      <td>-7</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15473</td>
      <td>15473</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25939/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25939/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25939/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour commented:
Failure is unrelated.
